app/assets/src/components/views/SampleUploadFlow/SampleUploadTable.jsx

```diff
@@ -0,0 +1,176 @@
+// Lists samples that the user has selected for upload.
+// Allows users to select and unselect samples, and remove unselected samples.
+
+import React from "react";
+import { difference, size, isEmpty, map, some } from "lodash/fp";
+import { SortDirection } from "react-virtualized";
+
+import PropTypes from "~/components/utils/propTypes";
+import { formatFileSize } from "~/components/utils/format";
+import { Table } from "~/components/visualizations/table";
+import { SELECT_ID_KEY } from "./constants";
+
+import cs from "./sample_upload_table.scss";
+
+const NAME_COLUMN = {
+  dataKey: "name",
+  width: 500,
+  label: "Sample Name",
+  className: cs.cell,
+  headerClassName: cs.header,
+};
+
+const BASESPACE_PROJECT_NAME_COLUMN = {
+  dataKey: "basespace_project_name",
+  flexGrow: 1,
+  width: 400,
+  label: "Basespace Project",
+  className: cs.cell,
+  headerClassName: cs.header,
+};
+
+const FILE_SIZE_COLUMN = {
+  dataKey: "file_size",
+  flexGrow: 1,
+  width: 100,
+  label: "File Size",
+  className: cs.cell,
+  headerClassName: cs.header,
+  cellRenderer: ({ cellData }) => formatFileSize(cellData),
+};
+
+const FILE_TYPE_COLUMN = {
+  dataKey: "file_type",
+  flexGrow: 1,
+  width: 200,
+  label: "File Type",
+  className: cs.cell,
+  headerClassName: cs.header,
+};
+
+const FILE_NAMES_COLUMN = {
+  dataKey: "file_names",
+  flexGrow: 1,
+  width: 100,
+  label: "Files",
+  className: cs.cell,
+  headerClassName: cs.header,
+  cellRenderer: ({ cellData }) => {
+    return (
+      <div>
+        {cellData.map(fileName => (
+          <div key={fileName} className={cs.fileName}>
+            {fileName}
+          </div>
+        ))}
+      </div>
+    );
+  },
+};
+
+export default class SampleUploadTable extends React.Component {
+  getColumns = () => {
+    const { sampleUploadType } = this.props;
+    if (sampleUploadType === "basespace") {
+      return [
+        NAME_COLUMN,
+        BASESPACE_PROJECT_NAME_COLUMN,
+        FILE_SIZE_COLUMN,
+        FILE_TYPE_COLUMN,
+      ];
+    }
+
+    if (sampleUploadType === "remote") {
```

Comment:
Why not join the two conditions?
If you do not like the `&&` you can go with `["remote", "local"].includes`